python/cusignal/_upfirdn.py

```diff
@@ -124,50 +124,10 @@ def _apply_1d(x, h_trans_flip, out, up, down, axis=-1):
 
 
 def _init_raw_apply1d_modules():
-    if '_raw_apply_1d_int' in _cached_modules:
```

Comment:
I think removing this will result in the raw kernel getting compiled on every call - I think it should remain but refer to the `_double` variant of the raw kernel.

Reply:
I'm not sure about that any more. Checkout the attached screenshot. It looks like the first call to init_modules takes 63ms, but all subsequent calls take ~500us. 
![cupy_initialization](https://user-images.githubusercontent.com/50021634/76458118-6ea62e80-63b0-11ea-8b8c-e652e3289150.png)

Reply:
Hey @gmarkall -- according to @mnicely, the int version of the raw kernel is 'orphaned' code. The double version is still being cached, i.e.

```python
module = cp.RawModule(code=loaded_from_source, options=("-std=c++11",))
    _cached_modules['_raw_apply_1d_double'] = \
        module.get_function("_raw_apply_1d_double")
```
so I *think* it's okay to get rid of the `_raw_apply_1d_int` entry in `_cached_modules`

Reply:
Ah nevermind. I see what you're saying. I'll add the `if` statement for completeness despite the fact that Matt's NSIGHT profile is showing that caching is seemingly taking place.

Reply:
@awthomp @gmarkall 
To clarify what you're looking at from the profile. There are 10 runs with Numba float64 kernels, then 10 runs with CuPy double kernels, then 10 runs with Numba float32 kernels, and then 10 runs with CuPy float kernels. 
It looks like CuPy kernels are getting cached like Numba's. 
I profiled a modified version this PR with CuPy NVTX markers.